data_pipeline/utils/pipeline_file_io.py

```diff
@@ -1,32 +1,153 @@
-from typing import Iterable
+import os
+from contextlib import ExitStack, contextmanager
+from typing import IO, Union, List
+from typing import Iterable, Callable
 import json
 
 import yaml
 
 
-def iter_write_jsonl_to_file(
-        json_list,
+class IntermediateBatchFileProcessing:
+    def __init__(
+            self,
+            intermediate_batch_processor: Callable,
+            batch_size: int,
+            truncate_file_after_processing: bool = True,
+            initial_record_processing_index: int = 0
+    ):
+        self.intermediate_batch_processor = intermediate_batch_processor
+        self.batch_size = batch_size
+        self.truncate_file_after_processing = truncate_file_after_processing
+        self.current_record_in_batch_index = initial_record_processing_index
+
+    def process_batch(
+            self, writer: IO,
+            process_file_instantly: bool = False
+    ):
+        if (
+                self.current_record_in_batch_index == self.batch_size
+                or process_file_instantly
+        ):
+            writer.flush()
+            self.intermediate_batch_processor(writer.name)
+            self.current_record_in_batch_index = 0
+
+            if self.truncate_file_after_processing:
+                writer.seek(0)
+                writer.truncate()
+        else:
+            self.current_record_in_batch_index += 1
+
+
+class WriteIterRecordsInBQConfig:
+    def __init__(
+            self,
+            bq_table: str = None,
+            file_location: str = None,
+            batch_processing: IntermediateBatchFileProcessing = None,
+            write_mode: str = 'a'
+    ):
+        self.bq_table = bq_table
+        self.batch_processing = batch_processing
+        self.file_location = file_location
+        self.write_mode = write_mode
+
+
+@contextmanager
+def get_opened_temp_file_for_tweet_response_types(
```

Comment:
is this still tweet specific?

Reply:
3 response types ( original tweet, retweet, and user), and they are all  have to be extracted from a  single nested retweet response

Reply:
Okay, I don't see a dependency on these types here. But if it was tweet specific then this should probably be moved to the twitter package?

Reply:
yes, you are right